ccl/oidcccl: TestOIDCEnabled failed

ccl/oidcccl.TestOIDCEnabled [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13057227?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13057227?buildTab=artifacts#/) on release-23.1 @ [e0983753ba31ec5f2f7f86071dbfd2c0e8b91e7f](https://github.com/cockroachdb/cockroach/commits/e0983753ba31ec5f2f7f86071dbfd2c0e8b91e7f):


```
=== RUN   TestOIDCEnabled
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/3ed8b395931fde05baf66bd2e4d88d97/logTestOIDCEnabled2727451157
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestOIDCEnabled
    authentication_oidc_test.go:248: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3ed8b395931fde05baf66bd2e4d88d97/logTestOIDCEnabled2727451157
--- FAIL: TestOIDCEnabled (6.22s)
=== RUN   TestOIDCEnabled/login_redirect
    authentication_oidc_test.go:242: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/oidcccl/authentication_oidc_test.go:242
        	Error:      	Not equal: 
        	            	expected: "some/random/path"
        	            	actual  : "/"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-some/random/path
        	            	+/
        	Test:       	TestOIDCEnabled/login_redirect
    --- FAIL: TestOIDCEnabled/login_redirect (0.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOIDCEnabled.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34284

--- a/pkg/ccl/oidcccl/authentication_oidc_test.go
+++ b/pkg/ccl/oidcccl/authentication_oidc_test.go
@@ -129,8 +129,8 @@ func TestOIDCEnabled(t *testing.T) {
 	sqlDB.Exec(t, `SET CLUSTER SETTING server.oidc_authentication.redirect_url = "https://cockroachlabs.com/oidc/v1/callback"`)
 	sqlDB.Exec(t, `set cluster setting server.oidc_authentication.claim_json_key = "email"`)
 	sqlDB.Exec(t, `set cluster setting server.oidc_authentication.principal_regex = '^([^@]+)@[^@]+$'`)
-	sqlDB.Exec(t, `SET CLUSTER SETTING server.oidc_authentication.enabled = "true"`)
 	sqlDB.Exec(t, fmt.Sprintf(`SET CLUSTER SETTING server.http.base_path = "%s"`, basePath))
+	sqlDB.Exec(t, `SET CLUSTER SETTING server.oidc_authentication.enabled = "true"`)
 
 	testCertsContext := s.NewClientRPCContext(ctx, username.TestUserName())
 	client, err := testCertsContext.GetHTTPClient()
@@ -203,7 +203,7 @@ func TestOIDCEnabled(t *testing.T) {
 		t.Fatalf("expected 307 status code but got: %d", resp.StatusCode)
 	}
 	if resp.Header.Get("Location") != basePath {
-		t.Fatalf("expected to be redirected to root")
+		t.Fatalf("expected to be redirected to %s", basePath)
 	}
 	foundCookie := false
 	for _, c := range resp.Cookies() {

